client.go

```diff
@@ -291,22 +292,20 @@ func (co *Conn) ReadMsgHeader(hdr *Header) ([]byte, error) {
 		p = make([]byte, length)
 		n, err = io.ReadFull(co.Conn, p)
 	}
+	p = p[:n]
 
 	if err != nil {
 		return nil, err
-	} else if n < headerSize {
+	} else if len(p) < headerSize {
 		return nil, ErrShortRead
 	}
 
-	p = p[:n]
-	if hdr != nil {
-		dh, _, err := unpackMsgHdr(p, 0)
-		if err != nil {
-			return nil, err
-		}
-		*hdr = dh
+	s := cryptobyte.String(p)
+	if hdr != nil && !hdr.unpack(&s) {
+		panic("dns: internal error: failed to unpack message header")
```

Comment:
i don't think we should introduce panics

I believe the prev. code was here to at least know something about the mangled packet, i.e the `*hdr = dh`

Reply:
So this panic actually can't ever happen (hence the "internal error" part). `Header.unpack` only ever returns false if p is shorter than 12 bytes, which is *exactly* what we check above with `if len(p) < headerSize`. I can document that.

Reply:
short comment can't hurt